fuzzer_runner/src/buttercup/fuzzer_runner/server.py

```diff
@@ -0,0 +1,235 @@
+    engine: str = Field(..., description="Fuzzing engine to use")
+    sanitizer: str = Field(..., description="Sanitizer to use")
+    timeout: int | None = Field(None, description="Timeout in seconds (overrides server default)")
+
+
+class MergeCorpusRequest(BaseModel):
+    corpus_dir: str = Field(..., description="Path to the corpus directory")
+    target_path: str = Field(..., description="Path to the target binary")
+    engine: str = Field(..., description="Fuzzing engine to use")
+    sanitizer: str = Field(..., description="Sanitizer to use")
+    output_dir: str = Field(..., description="Output directory for merged corpus")
+    timeout: int | None = Field(None, description="Timeout in seconds (overrides server default)")
+
+
+class FuzzResponse(BaseModel):
+    task_id: str = Field(..., description="Unique task identifier")
+    status: str = Field(..., description="Task status")
+    result: dict[str, Any] | None = Field(None, description="Fuzzing result if completed")
+    error: str | None = Field(None, description="Error message if failed")
+
+
+class MergeCorpusResponse(BaseModel):
+    task_id: str = Field(..., description="Unique task identifier")
+    status: str = Field(..., description="Task status")
+    error: str | None = Field(None, description="Error message if failed")
+
+
+class HealthResponse(BaseModel):
+    status: str = Field(..., description="Server health status")
+    version: str = Field(..., description="Server version")
+
+
+@app.get("/health", response_model=HealthResponse)
+async def health_check() -> HealthResponse:
+    """Health check endpoint"""
+    return HealthResponse(status="healthy", version="0.1.0")
+
+
+@app.post("/fuzz", response_model=FuzzResponse)
+async def run_fuzzer(request: FuzzRequest, background_tasks: BackgroundTasks) -> FuzzResponse:
+    """Run a fuzzer with the given configuration"""
+    task_id = str(uuid.uuid4())
+
+    # Validate paths
+    if not Path(request.corpus_dir).exists():
+        raise HTTPException(status_code=400, detail=f"Corpus directory does not exist: {request.corpus_dir}")
+
+    if not Path(request.target_path).exists():
+        raise HTTPException(status_code=400, detail=f"Target path does not exist: {request.target_path}")
+
+    # Create fuzz configuration
+    fuzz_conf = FuzzConfiguration(
+        corpus_dir=request.corpus_dir,
+        target_path=request.target_path,
+        engine=request.engine,
+        sanitizer=request.sanitizer,
+    )
+
+    # Store task info
+    active_tasks[task_id] = {
+        "type": "fuzz",
+        "status": "running",
+        "config": fuzz_conf,
+        "timeout": request.timeout or (server_settings.timeout if server_settings else 1000),
+    }
+
+    # Run fuzzer in background
+    background_tasks.add_task(_run_fuzzer_task, task_id, fuzz_conf, request.timeout)
+
+    return FuzzResponse(task_id=task_id, status="running")
+
+
+@app.post("/merge-corpus", response_model=MergeCorpusResponse)
```

Comment:
Is there anyway of making the paths into constants shared by the proxy and runner?